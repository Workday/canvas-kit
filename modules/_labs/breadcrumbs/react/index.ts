import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
