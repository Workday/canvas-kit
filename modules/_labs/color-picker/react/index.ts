export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
