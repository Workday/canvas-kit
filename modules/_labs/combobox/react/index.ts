import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
