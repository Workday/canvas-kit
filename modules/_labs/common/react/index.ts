export * from './lib/Flex';
export * from './lib/Box';
