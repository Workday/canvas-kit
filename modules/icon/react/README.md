# Canvas Kit Icon

Components assisting the rendering of Canvas accent, applet, and system icons.

A Workday asset package must be used with these components. No icons are bundled with this package.

To browse a full list of icons, take a look at the
[Canvas Design System resources](https://design.workday.com/resources) page.

## Installation

```sh
yarn add @workday/canvas-kit-react
```

or

```sh
yarn add @workday/canvas-kit-react-icon
```

### Design Assets

Install one of the following icon packages:

```sh
@workday/canvas-accent-icons-web
@workday/canvas-applet-icons-web
@workday/canvas-system-icons-web
```

# Accent Icons

## Usage

Use with `@workday/canvas-accent-icons-web`.

```tsx
import { colors } from '@workday/canvas-kit-react-core'
import { AccentIcon } from '@workday/canvas-kit-react-icon'
import { shieldIcon } from '@workday/canvas-accent-icons-web'

<AccentIcon icon={shieldIcon} />
<AccentIcon icon={shieldIcon} color={colors.pomegranate500} />
<AccentIcon icon={shieldIcon} size={80} />
```

## Static Properties

> None

## Component Props

### Required

#### `icon: CanvasIcon`

> Icon to display from `@workday/canvas-accent-icons-web`.

### Optional

#### `color: string`

> Icon color from `@workday/canvas-colors-web`.

Default: `colors.blueberry500`

---

#### `size: number`

> Size of the icon.

Default: `56`

---

#### `transparent: boolean`

> Toggle for transparent accent icon background. If false, the background fill will be white.

Default: `false`

---

#### `iconRef: React.Ref<HTMLSpanElement>`

> Returns the ref to the rendered icon.

---

# Applet Icons

## Usage

Use with `@workday/canvas-applet-icons-web`.

```tsx
import { AppletIcon } from '@workday/canvas-kit-react-icon'
import { benefitsIcon } from '@workday/canvas-applet-icons-web'

<AppletIcon icon={benefitsIcon} />
<AppletIcon icon={benefitsIcon} color={AppletIcon.Colors.Pomegranate} />
<AppletIcon icon={benefitsIcon} size={60} />
```

## Static Properties

#### `Colors: BrandingColor`

> An enum of the various Canvas hues (`Pomegranate`, `Blueberry`, `Cinnamon`, etc.).

```tsx
<AppletIcon icon={benefitsIcon} color={AppletIcon.Colors.Pomegranate} />
```

- `Cinnamon`
- `Peach`
- `ChiliMango`
- `Cantaloupe`
- `SourLemon`
- `JuicyPear`
- `Kiwi`
- `GreenApple`
- `Watermelon`
- `Jewel`
- `Toothpaste`
- `Blueberry`
- `Plum`
- `BerrySmoothie`
- `Blackberry`
- `IslandPunch`
- `GrapeSoda`
- `Pomegranate`
- `FruitPunch`
- `RootBeer`
- `ToastedMarshmallow`
- `Cappuccino`
- `Licorice`
- `BlackPepper`

## Component Props

### Required

#### `icon: CanvasIcon`

> Icon to display from `@workday/canvas-applet-icons-web`.

### Optional

#### `color: AppletIcon.Colors`

> Icon color hue. Must use a member of the `AppletIcon.Colors` static enum.

Default: `AppletIcon.Colors.Blueberry`

---

#### `size: number`

> Size of the icon in `px`.

Default: `92`

---

#### `iconRef: React.Ref<HTMLSpanElement>`

> Returns the ref to the rendered icon.

---

# System Icons

## Usage

Use with `@workday/canvas-system-icons-web`.

```tsx
import { colors } from '@workday/canvas-kit-react-core'
import { SystemIcon } from '@workday/canvas-kit-react-icon'
import { activityStreamIcon } from '@workday/canvas-system-icons-web'

<SystemIcon icon={activityStreamIcon} />
<SystemIcon icon={activityStreamIcon} color={colors.blueberry500} />
<SystemIcon
  icon={activityStreamIcon}
  accent={colors.frenchVanilla100}
  fill={colors.blueberry500}
  background={colors.blueberry500}
/>
<SystemIcon icon={activityStreamIcon} size={48} />
```

## Static Properties

> None

## Component Props

### Required

#### `icon: CanvasIcon`

> Icon to display from `@workday/canvas-system-icons-web`

### Optional

#### `size: number`

> Size of the icon in `px`.

Default: `24`

---

#### `color: string`

> Icon color. This will define `accent` and `fill`. `accent` and `fill` will override this property
> if defined.

Default: `iconColors.standard`

---

#### `colorHover: string`

> Hover color of the icon. This will define `accentHover` and `fillHover`.

Default: `iconColors.hover`

---

#### `accent: string`

> `.wd-icon-accent` color. This will override `color`.

---

#### `accentHover: string`

> `.wd-icon-accent` color on hover. This will override `colorHover`.

---

#### `background: string`

> `.wd-icon-background` color.

Default: `'transparent'`

---

#### `backgroundHover: string`

> `.wd-icon-background` color on hover.

Default: `'transparent'`

---

#### `fill: string`

> `.wd-icon-fill` color. This will override `color`.

---

#### `fillHover: string`

> `.wd-icon-fill` color on hover. This will override `colorHover`.

---

#### `iconRef: React.Ref<HTMLSpanElement>`

> Returns the ref to the rendered icon.

---

# System Icon Circle

A system icon with a colored circular background. Icon color will be determined based on contrast
against the background color provided.

## Usage

Use with `@workday/canvas-accent-icons-web`.

```tsx
import { colors } from '@workday/canvas-kit-react-core'
import { SystemIconCircle } from '@workday/canvas-kit-react-icon'
import { shieldIcon } from '@workday/canvas-accent-icons-web'

<SystemIconCircle icon={shieldIcon} />
<SystemIconCircle icon={shieldIcon} background={colors.pomegranate500} />
```

## Static Properties

> None

## Component Props

### Required

#### `icon: CanvasIcon`

> Icon to display from `@workday/canvas-accent-icons-web`.

### Optional

#### `background: string`

> Background color from `@workday/canvas-colors-web`.

Default: `colors.soap300`

---

#### `size: SystemIconCircleSize | number`

> Size of the icon.

Default: `SystemIconCircleSize.l` (`40`)

---

#### `iconRef: React.Ref<HTMLSpanElement>`

> Returns the ref to the rendered icon.

---

# Graphics

## Usage

Use with `@workday/canvas-graphics-web`.

```tsx
import { colors } from '@workday/canvas-kit-react-core'
import { Graphic } from '@workday/canvas-kit-react-icon'
import { badgeAchievementGraphic } from '@workday/canvas-graphics-web'

<Graphic src={badgeAchievementGraphic} />
<Graphic src={badgeAchievementGraphic} width={80}/>
<Graphic src={badgeAchievementGraphic} height={80}/>
<Graphic src={badgeAchievementGraphic} grow={true} />
```

## Static Properties

#### `Size: SystemIconCircleSize`

> The size of the circle background. The icon will scale to fit.

| Size | Pixels |
| ---- | ------ |
| xs   | 16     |
| s    | 24     |
| m    | 32     |
| l    | 40     |
| xl   | 64     |
| xxl  | 120    |

## Component Props

### Required

#### `src: CanvasGraphic`

> Graphic to display from `@workday/canvas-graphics-web`

### Optional

#### `width: number | string`

> Graphic width in `px`. `width` takes precedence over `height` in order to preserve the graphic's
> ratio.

Default: `width of graphic`

---

#### `height: number | string`

> Graphic height in `px`. If set, `width` will be set to `100%`.

Default: `height of graphic`

---

#### `grow: boolean`

> Expand graphic to fit container. `grow` takes precedence over both `width` and `height`.

Default: `false`

#### `iconRef: React.Ref<HTMLSpanElement>`

> Returns the ref to the rendered icon.

---
