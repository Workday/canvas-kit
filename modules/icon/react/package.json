{
  "name": "@workday/canvas-kit-react-icon",
  "version": "3.5.0",
  "description": "Canvas icons",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "http://github.com/Workday/canvas-kit/tree/master/modules/icon/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts",
    "ts3.5/**/*"
  ],
  "typesVersions": {
    "<=3.5": {
      "*": [
        "ts3.5/*"
      ]
    }
  },
  "scripts": {
    "watch": "yarn build:es6 -w",
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build:downlevel-dts": "yarn run downlevel-dts dist ts3.5/dist",
    "build": "npm-run-all --parallel build:cjs build:es6 --sequential build:downlevel-dts",
    "depcheck": "node ../../../utils/check-dependencies-exist.js"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "icon"
  ],
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  },
  "dependencies": {
    "@emotion/core": "^10.0.28",
    "@emotion/is-prop-valid": "^0.8.2",
    "@emotion/styled": "^10.0.27",
    "@workday/canvas-kit-react-common": "^3.5.0",
    "@workday/canvas-kit-react-core": "^3.5.0",
    "@workday/design-assets-types": "^0.2.4"
  },
  "devDependencies": {
    "@workday/canvas-accent-icons-web": "^1.0.0",
    "@workday/canvas-applet-icons-web": "^0.17.50",
    "@workday/canvas-system-icons-web": "^1.0.20"
  }
}
