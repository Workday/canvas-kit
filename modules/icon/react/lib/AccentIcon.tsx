import * as React from 'react';
import {colors} from '@workday/canvas-kit-react-core';
import {CanvasAccentIcon, CanvasIconTypes} from '@workday/design-assets-types';
import {CSSObject} from '@emotion/core';
import Icon, {IconProps} from './Icon';

export interface AccentIconStyles {
  /**
   * The fill color of the AccentIcon.
   * @default colors.blueberry500
   */
  color?: string;
  /**
   * If true, set the background fill of the AccentIcon to `transparent`. If false, set the background fill of the AccentIcon to `colors.frenchVanilla100`.
   * @default false
   */
  transparent?: boolean;
}

export interface AccentIconProps extends AccentIconStyles, Omit<IconProps, 'src' | 'type'> {
  /**
   *  The icon to display from `@workday/canvas-accent-icons-web`.
   */
  icon: CanvasAccentIcon;
  /**
   * The size of the AccentIcon in `px`.
   * @default 56
   */
  size?: number;
}

export const accentIconStyles = ({
  color = colors.blueberry500,
  transparent = false,
}: AccentIconStyles): CSSObject => ({
  '& .color-500': {
    fill: color,
  },
  '& .french-vanilla-100': {
    fill: transparent ? 'transparent' : colors.frenchVanilla100,
  },
});

export default class AccentIcon extends React.Component<AccentIconProps> {
  render() {
    const {transparent = false, size = 56, icon, color, iconRef, ...elemProps} = this.props;

    return (
      <Icon
        src={icon}
        type={CanvasIconTypes.Accent}
        styles={accentIconStyles({color, transparent})}
        size={size}
        iconRef={iconRef}
        {...elemProps}
      />
    );
  }
}
