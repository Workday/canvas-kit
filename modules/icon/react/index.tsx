import AccentIcon from './lib/AccentIcon';
import AppletIcon from './lib/AppletIcon';
import SystemIcon from './lib/SystemIcon';
import SystemIconCircle from './lib/SystemIconCircle';
import Graphic from './lib/Graphic';
import Svg from './lib/Svg';

export {AccentIcon, AppletIcon, SystemIcon, SystemIconCircle, Graphic, Svg};

export * from './lib/AccentIcon';
export * from './lib/AppletIcon';
export * from './lib/SystemIcon';
export * from './lib/SystemIconCircle';
export * from './lib/Graphic';
export * from './lib/Svg';
