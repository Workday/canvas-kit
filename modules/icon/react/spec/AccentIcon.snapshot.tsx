import * as renderer from 'react-test-renderer';
import AccentIcon from '../lib/AccentIcon';
import * as React from 'react';
import {shieldIcon} from '@workday/canvas-accent-icons-web';

describe('Accent Icon Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<AccentIcon icon={shieldIcon} />);
    expect(component).toMatchSnapshot();
  });
});
