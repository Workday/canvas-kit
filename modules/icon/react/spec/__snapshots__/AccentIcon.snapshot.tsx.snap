// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accent Icon Snapshots renders as expected 1`] = `
.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .color-500 {
  fill: #005cb9;
}

.emotion-0 .french-vanilla-100 {
  fill: #ffffff;
}

<span
  className="emotion-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"56\\" height=\\"56\\" class=\\"wd-accent-shield wd-accent\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 56 56\\"><g fill=\\"none\\" fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill=\\"#005DBA\\" fill-rule=\\"nonzero\\" d=\\"M9.011 11.573a.807.807 0 0 1 .537-.746l17.904-5.654a2.114 2.114 0 0 1 1.096 0l17.904 5.654a.813.813 0 0 1 .537.764S46.9 20.903 45 31c-1.9 10.097-12.663 15.503-17 20-3.989-4.497-15.1-9.568-17-20-1.9-10.432-1.989-19.427-1.989-19.427zM27.946 7.08c.066.016.077.01.108 0l-17.022 5.376.021.61c.047 1.235.123 2.633.235 4.163.32 4.374.862 8.925 1.68 13.413.825 4.531 3.538 8.178 8.385 12.214.668.556 1.36 1.11 2.25 1.803.114.09 1.728 1.342 2.179 1.697a46.566 46.566 0 0 1 2.284 1.893c2.413-2.068 6.486-5.296 6.384-5.21 4.952-4.173 7.735-7.895 8.585-12.409.816-4.336 1.358-8.848 1.677-13.265a120.584 120.584 0 0 0 .257-4.909L27.946 7.08zm-13.897 8.463a.742.742 0 0 1 .466-.727l12.97-4.632a1.78 1.78 0 0 1 1.03 0l12.97 4.632a.755.755 0 0 1 .467.731S41.4 22.313 40 30c-1.4 7.687-8.804 11.577-12 15-2.94-3.423-10.6-7.058-12-15-1.4-7.942-1.95-14.457-1.95-14.457zm1.98.751l.02.202a183.943 183.943 0 0 0 1.823 13.174c.602 3.41 2.475 6.083 5.846 9.043.457.401.932.802 1.543 1.304.09.074 1.2.982 1.513 1.242.49.407.909.766 1.277 1.098a209.19 209.19 0 0 1 4.067-3.482c3.461-3.07 5.39-5.807 6.011-9.216.605-3.32 1.097-6.712 1.484-9.99a156.596 156.596 0 0 0 .362-3.373L28 12.019l-11.97 4.275zm2.98 2.844a1 1 0 0 1 1.981-.276c.977 6.985 1.961 11.252 2.844 12.586a1 1 0 1 1-1.668 1.104c-1.117-1.69-2.133-6.089-3.156-13.414zM25 36a1 1 0 1 1 0-2 1 1 0 0 1 0 2z\\" class=\\"color-500\\"/><path fill=\\"#FFF\\" d=\\"M27.946 7.08l17.023 5.376-.022.674a120.584 120.584 0 0 1-.235 4.235c-.32 4.417-.861 8.93-1.677 13.265-.85 4.514-3.633 8.236-8.585 12.408.102-.085-3.97 3.143-6.384 5.211a46.566 46.566 0 0 0-2.284-1.893c-.45-.355-2.065-1.607-2.18-1.697a88.926 88.926 0 0 1-2.249-1.803c-4.847-4.036-7.56-7.683-8.385-12.214-.818-4.488-1.36-9.04-1.68-13.413a119.04 119.04 0 0 1-.256-4.773L28.054 7.08c-.03.01-.042.016-.108 0zm-13.897 8.463S14.6 22.058 16 30c1.4 7.942 9.06 11.577 12 15 3.196-3.423 10.6-7.313 12-15 1.4-7.687 1.952-14.453 1.952-14.453a.755.755 0 0 0-.467-.73l-12.97-4.633a1.78 1.78 0 0 0-1.03 0l-12.97 4.632a.742.742 0 0 0-.466.727zm13.826-3.568l.037.013-.037-.013zm-11.846 4.32L28 12.018l11.975 4.277-.024.244c-.09.93-.203 1.98-.338 3.128-.387 3.279-.879 6.67-1.484 9.991-.62 3.409-2.55 6.146-6.011 9.216a209.19 209.19 0 0 0-4.067 3.482 40.906 40.906 0 0 0-1.277-1.098c-.313-.26-1.423-1.168-1.513-1.242a62.262 62.262 0 0 1-1.543-1.304c-3.37-2.96-5.244-5.633-5.845-9.043a183.943 183.943 0 0 1-1.844-13.376zm2.98 2.843c1.024 7.325 2.04 11.725 3.157 13.414a1 1 0 1 0 1.668-1.104c-.883-1.334-1.867-5.601-2.844-12.586a1 1 0 0 0-1.98.276zM25 36a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\" class=\\"french-vanilla-100\\"/></g></svg>",
    }
  }
/>
`;
